src/Elastic.Apm/Metrics/MetricsProvider/BreakdownMetricsProvider.cs

```diff
@@ -28,7 +43,6 @@ internal class BreakdownMetricsProvider : IMetricsProvider
 		private bool _loggedWarning;
 
 		private readonly object _lock = new();
```

Comment:
I know this part was unchanged, but a comment (or variable naming) on which field(s) is/are protected by that `lock` could be helpful.
Maybe considering a thread-safe collection would also be worthwhile here?

Reply:
Very good point on naming - I changed it and also added a comment.

As mentioned over zoom, a thread-safe collection would not be enough, since the unit of operation is a bit more than reading/writing from/to the collection - e.g. in `GetSamples` we need to clear the collection as well in the same round. So we keep the lock with better naming.